packages/e2e/cypress/integration/explore.spec.ts

```diff
@@ -38,18 +38,79 @@ describe('Explore', () => {
+        // FIXME disabling save changes button is currently broken...
+        // cy.findByText('Save changes').parent().should('be.disabled');
 
-        // TODO introduce more changes
+        cy.findAllByText('Loading chart').should('have.length', 0); // Finish loading Wait until finish loading
         cy.findByText('Column chart').click(); // Change chart type
         cy.findByText('Bar chart').click();
 
-        cy.findByText('Save changes').parent().should('not.be.disabled');
+        // cy.findByText('Save changes').parent().should('not.be.disabled');
         cy.findByText('Save changes').parent().click();
 
         cy.findByText('Success! Chart was saved.');
     });
 
+    it('Should change chart config type', () => {
+        cy.visit(`/projects/${SEED_PROJECT.project_uuid}/tables`);
+
+        cy.findByText('Orders').click();
+        cy.findByText('First name').click();
+        cy.findByText('Unique order count').click();
+        cy.get('th b').first().should('have.text', 'First name').click();
+        cy.get('td', { timeout: 10000 }).eq(1).should('have.text', 'Aaron');
+
+        cy.findByText('Charts').prev().click(); // open chart
+
+        cy.findByText('Customers First name');
+
+        cy.findByText('Column chart').click(); // Change chart type
+        cy.findByText('Bar chart').click();
+
+        cy.findByText('Customers First name');
+
+        cy.findByText('Bar chart').click();
+        cy.findByText('Line chart').click();
+
+        cy.findByText('Customers First name');
+
+        cy.findByText('Line chart').click();
+        cy.findByText('Scatter chart').click();
+
+        cy.findByText('Customers First name');
+
+        cy.findByText('Scatter chart').click();
+        cy.get('[name="Table"]').click();
+
+        cy.findByText('Customers first name'); // Different label ¿?
+
+        cy.get('.bp3-icon-panel-table').first().parent().click();
+        cy.findByText('Big number').click();
+
+        cy.findByText('Orders Unique order count'); // Different label ¿?
+    });
+
+    it('Should change chart config layout', () => {
+        cy.visit(`/projects/${SEED_PROJECT.project_uuid}/tables`);
+
+        cy.findByText('Orders').click();
+        cy.findByText('First name').click();
+        cy.findByText('Unique order count').click();
+        cy.get('th b').first().should('have.text', 'First name').click();
+        cy.get('td', { timeout: 10000 }).eq(1).should('have.text', 'Aaron');
+
+        cy.findByText('Charts').prev().click(); // open chart
+
+        cy.findByText('Customers First name');
+        cy.get('g').children('text').should('have.length.lessThan', 30); // without labels
+
+        cy.findByText('Configure').click();
+        cy.findByText('Series').click();
+        cy.findByText('Value labels');
+        cy.get('option[value="top"]').parent().select('top');
+
+        cy.get('g').children('text').should('have.length.greaterThan', 30); // with labels
+    });
```

Comment:
We can keep this test but I'm not the biggest fan of these assertions since they are too abstract and since this is an external library we need to assume they are rendering correctly. 

I would prefer for us to have unit tests on the chart configuration components/hooks to make sure the echarts configuration object has the correct properties and values.

Reply:
I agree, this is the easier way/ more stable to test that `text` fields are appearing on the chart , without breaking it. I also agree about the unit test, I'm just doing this here because recently a chart broke when labels were included. My new rule is : if there was a critical bug on master, fix it and write a unit test to avoid that from happening. 

I'll keep refactoring those tests if needed, until we find a way to make them stable in the long run.